### PR TITLE
03 food list

### PR DESCRIPTION
**Pull request by Alejandroq12**
Hello, team. I made sure to fulfill the following requirements:

### Food list:

- Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- Should lead to a form that allows users to add new food.

image

image

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive management system for food items, including listing, creating, editing, updating, and deleting foods.
	- Added user-specific management capabilities for food items.
	- Implemented a new relationship model between recipes and foods.
	- Enhanced recipe model with validation for public visibility.
- **Enhancements**
	- Improved user model with name validation and updated associations for recipes and foods.
- **UI Updates**
	- Added new views and forms for food item management, including creation, editing, and listing of food items.
- **Routing**
	- Established new routes for managing food resources.
- **Database**
	- Created database migrations for food items and their associations with recipes.
- **Testing**
	- Added test cases for food management functionalities.

### CODE REVIEW

File: test/controllers/foods_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  test 'should get new' do
+    get foods_new_url
+    assert_response :success
+  end
+
+  test 'should get create' do
+    get foods_create_url
+    assert_response :success
+  end
```
[the remark this answers is not in the document]
Comment (Alejandroq12) [reply test/controllers/foods_controller_test.rb:22]: I will implement tests at the last milestone. But, thanks.

File: test/controllers/foods_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  test 'should get edit' do
+    get foods_edit_url
+    assert_response :success
+  end
+
+  test 'should get update' do
+    get foods_update_url
+    assert_response :success
+  end
```
[the remark this answers is not in the document]
Comment (Alejandroq12) [reply test/controllers/foods_controller_test.rb:32]: I will implement tests at the last milestone. But, thanks.

File: test/controllers/foods_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  test 'should get destroy' do
+    get foods_destroy_url
+    assert_response :success
+  end
```
[the remark this answers is not in the document]
Comment (Alejandroq12) [reply test/controllers/foods_controller_test.rb:37]: I will implement tests at the last milestone. But, thanks.

**Comment by Nessrine88** [comment]
### STATUS APPROVED 🟢 
Great implementing the requirements:
✅ Add new food
✅ Display a list food
✅ Linters are passed

it's time to merge it!!

**Pull request merged**